forum/startups/views.py

```diff
@@ -101,6 +147,90 @@ def destroy(self, request, *args, **kwargs):
+            return Response({"detail": "Startup not found."}, status=status.HTTP_404_NOT_FOUND)
+        except ValidationError as e:
+            return Response({"detail": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+        except IntegrityError:
+            return Response({"detail": "Database integrity error occurred."},
+                        status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            return Response({"detail": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def list(self, request, *args, **kwargs):
+        """
+        List all Startup instances.
+        
+        This method retrieves a list of all startups and handles any errors that
+        may occur during retrieval.
+
+        Args:
+            request (Request): The request object to list startups.
+
+        Returns:
+            Response: A response containing the list of startups or error details.
+        """
+        try:
+            queryset = self.get_queryset()
+            serializer = self.get_serializer(queryset, many=True)
+            return Response(serializer.data)
+        except ValidationError as e:
+            return Response({"detail": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+        except IntegrityError:
+            return Response({"detail": "Database integrity error occurred."},
+                        status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            return Response({"detail": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        
+    def get_queryset(self):
+        """
+        Retrieves the queryset of startup profiles based on user role and query parameters.
+        Search and filtering options are provided. Custom error handling is included to 
+        manage permission issues.
+
+        Returns:
+            QuerySet: Filtered and searchable queryset for the list and retrieve actions.
+
+        Raises:
+            PermissionDenied: If the user is not authenticated or does not have the 'investor' role.
+        """
+        if self.action in ['list', 'retrieve']:
+            if not self.request.user.is_authenticated or self.request.user.active_role.name != 'investor':
+                raise PermissionDenied("You do not have permission to view this data.")
+        
+        queryset = self.queryset
+        search = self.request.query_params.get('search', None)
+        if search:
+            queryset = queryset.filter(company_name__icontains=search)
+
+        company_name = self.request.query_params.get('company_name', None)
+        industry = self.request.query_params.get('industry', None)
+
+        if company_name:
+            queryset = queryset.filter(company_name__icontains=company_name)
+        industry_name = self.request.query_params.get('industry_name', None)
+        if industry_name:
+            queryset = queryset.filter(industries__name__icontains=industry_name)
+
+        return queryset
```

Comment:
`get_queryset` has multiple checks for different query parameters, making the method lengthy.
Separate filtering logic into a dedicated helper method for readability and scalability.
```
def filter_queryset(self, queryset):
    search = self.request.query_params.get('search')
    company_name = self.request.query_params.get('company_name')
    industry_name = self.request.query_params.get('industry_name')

    if search:
        queryset = queryset.filter(company_name__icontains=search)
    if company_name:
        queryset = queryset.filter(company_name__icontains=company_name)
    if industry_name:
        queryset = queryset.filter(industries__name__icontains=industry_name)

    return queryset

def get_queryset(self):
    queryset = super().get_queryset()
    return self.filter_queryset(queryset)
```